src/Babylonian.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.16;
+
+// Based on https://github.com/Uniswap/solidity-lib/blob/c01640b0f0f1d8a85cba8de378cc48469fcfd9a6/contracts/libraries/Babylonian.sol
+
+// computes square roots using the babylonian method
+// https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Babylonian_method
+library Babylonian {
+    // credit for this implementation goes to
+    // https://github.com/abdk-consulting/abdk-libraries-solidity/blob/master/ABDKMath64x64.sol#L687
+    function sqrt(uint256 x) internal pure returns (uint256) {
+        unchecked {
+            if (x == 0) return 0;
+            // this block is equivalent to r = uint256(1) << (BitMath.mostSignificantBit(x) / 2);
+            // however that code costs significantly more gas
+            uint256 xx = x;
+            uint256 r = 1;
+            if (xx >= 0x100000000000000000000000000000000) {
```

Comment:
nit: the formatting in the updated file https://github.com/abdk-consulting/abdk-libraries-solidity/blob/5e1e7c11b35f8313d3f7ce11c1b86320d7c0b554/ABDKMath64x64.sol#L725 is a bit more compact and readable

Reply:
Right, but not sure it's worth the change tbh.
I think the Uniswap version which is a bit easier to diff (as only that file exists in the source).

Reply:
Fine to keep the Uni version. Resolving.